### DISCUSSION TITLE
Toy ECCO

### DISCUSSION THREAD

**Discussion by Shreyas911** [ECCO-Hackweek/ecco-2024#159] [General]
Project discussion here. :)

**Comment by gaelforget** [comment]
tentative list of next steps : 

- [x] wire Enzyme/autodiff into Optim.jl
- [x] same as before but with bulk formulae
- [x] design cost function
- [x] add glacier model example
- [x] starting with scalar function (hl)
- [ ] check bulkformual against fortarn code
- [ ] plot hl in phase space (heatmap)
- [ ] do the same with adjoint variables (heatmap)
- [ ] create repo into hackweek of the template

**Comment by gaelforget** [comment]
https://github.com/ECCO-Hackweek/ecco-2024/issues/161

**Comment by eldavenport** [comment]
Oct 16: 

- Working optimization problem with Enzyme applied to AirSeaFlux::bulkformulae. Using some fake observations to find initial conditions that get the appropriate output. Applied to 4 variables (hl,hs,evap,tau).
- Starting L96 version of the above (to move to a more dynamical systems-like example)
- Mountain glacier model working with tapenade, JAX, and PyTorch (struggling with Enzyme)

Miscellaneous update:
- Tapenade works directly on p-cluster without Docker!

**Comment by trostelm** [comment]
![Toy ECCO Slide](https://github.com/user-attachments/assets/41cb2012-9dc9-485a-b546-0c52cf999b5b)

**Comment by gaelforget** [comment]
- Added examples for ForwardDiff.jl and Zygote.jl (now in [ECCO.jl](https://gaelforget.github.io/ECCO.jl/dev/) `v0.1.1`) in [this notebook](https://github.dev/ECCO-Hackweek/EH24-ToyECCO/blob/main/notebooks/ForwardDiff_Zygote_Enzyme.ipynb) 
- added a list of content in [the README](https://github.com/ECCO-Hackweek/EH24-ToyECCO?tab=readme-ov-file#summary-slide)
